h3py/util.pyx

```diff
@@ -0,0 +1,135 @@
+from libc cimport stdlib
+from cython.view cimport array
+
+from h3py.libh3 cimport H3int, H3str
+cimport h3py.libh3 as h3c
+
+# todo: should we use C API functions instead? (stringToH3 and h3ToString)
+cpdef H3int hex2int(H3str h):
+    return int(h, 16)
+
+cpdef H3str int2hex(H3int x):
+    return hex(x)[2:]
+
+
+class H3ValueError(ValueError):
+    pass
+
+class InvalidH3Address(H3ValueError):
+    pass
+
+class InvalidH3Edge(H3ValueError):
+    pass
+
+class InvalidH3Resolution(H3ValueError):
+    pass
+
+cdef check_addr(H3int h):
+    if h3c.h3IsValid(h) == 0:
+        raise InvalidH3Address(h)
+
+cdef check_edge(H3int e):
+    if h3c.h3UnidirectionalEdgeIsValid(e) == 0:
+        raise InvalidH3Edge(e)
+
+cdef check_res(int res):
+    if res < 0 or res > 15:
+        raise InvalidH3Resolution(res)
+
+
+
+## todo: can i turn these two into a context manager?
+cdef H3int* create_ptr(size_t n) except *:
+    if n <= 0:
+        # raise H3ValueError("Can't allocate an array of length 0!")
+        # actually, n=0 should be OK.
+        pass
```

Comment:
What does it return if `n == 0`? Seems like it should just pass it along to `calloc`

Reply:
it does indeed just pass it along to `calloc`. I'll remove that block to avoid confusion.